ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetMinerDataByBlockHash.java

```diff
@@ -82,19 +83,16 @@ public static MinerDataResult createMinerDataResult(
     final ProtocolSpec protocolSpec = protocolSchedule.getByBlockHeader(blockHeader);
     final Wei staticBlockReward = protocolSpec.getBlockReward();
     final Wei transactionFee =
-        block.getTransactions().stream()
+        blockchainQueries
+            .transactionReceiptsByBlockHash(blockHeader.getHash(), protocolSchedule)
```

Comment:
Is there any reason why you didn't use DefaultBlockchain.getTxReceipts here ?

Reply:
1. `DefaultBlockchain.getTxReceipts` does not conform to the current API of `BlockchainQueries` class
2. More importantly, `blockchainQueries.transactionReceiptsByBlockHash` returns a list of  `TransactionReceiptWithMetadata` instead of `TransactionReceipt`, which is required for further processing.

Reply:
1. You can get the defaultBlockchain instance from BlockchainQueries (BlockchainQueries.getblockchain())
2. Right, you will need to iterate over the result and create a TransactionReceiptWithMetadata from the transactionReceipt with same logic as transactionReceiptByTransactionHash